<a href="https://colab.research.google.com/github/sergioGarcia91/SeismicUP/blob/main/03_Analisis_de_subcatalogos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

En este notebook se procederá con el análisis del catálogo sísmico dividiéndolo en tres subcatalogos:

1. `FC` (Full Catalog): corresponde a todos los eventos descargados.
2. `BSN` (Bucaramanga Seismic Nest Catalog): corresponde a los eventos asociados al área de estudio de García-Arias et al. (2026).
3. `CC` (Campo Colorado Catalog): corresponde a los eventos contenidos en un área de 100x100 km, centrada en el punto de referencia del Campo Colorado (X = 4'919.222,973; Y = 2'308.553,476; EPSG:9377).

A continuación se presentan los límites espaciales de cada catálogo expresados en coordenadas geográficas EPSG:4326:

| Catálogo | Longitud (min, max) | Latitud (min, max) |
| :------: | :-----------------: | :----------------: |
|    FC    |    -76.00, -71.00   |     3.00, 11.00    |
|    BSN   |    -73.40, -72.80   |     6.50, 7.10     |
|    CC    |    -74.19, -73.26   |     6.32, 7.25     |






# Inicio

In [ ]:
!git clone https://github.com/sergioGarcia91/SeismicUP.git

Cloning into 'SeismicUP'...
remote: Enumerating objects: 529, done.
remote: Counting objects: 100% (228/228), done.
remote: Compressing objects: 100% (226/226), done.
remote: Total 529 (delta 134), reused 0 (delta 0), pack-reused 301 (from 1)
Receiving objects: 100% (529/529), 23.17 MiB | 18.38 MiB/s, done.
Resolving deltas: 100% (282/282), done.


In [ ]:
# para incluir la libreria clonada
import sys
sys.path.append("/content/SeismicUP")

import seismicup as sup

In [ ]:
!pip -q install python-calamine

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 935.1/935.1 kB 13.7 MB/s eta 0:00:00


In [ ]:
!pip3 install contextily

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import geopandas as gpd
import seaborn as sns
import os
import re
import contextily as cx #para el basemap en geopandas
import xyzservices.providers as xyz #para escoger el basemap
import seismicup as sup
import urllib.request
import matplotlib.font_manager as fm
import random

from matplotlib.colors import LogNorm # para la escala logaritmica de los colores
from matplotlib.ticker import LogFormatter, LogLocator # escala log
from scipy import stats # regresion lineal

# Conectar al Drive

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Cambiar Fuente

In [ ]:
sup.plots.get_TimesNewRoman_font()

# Paths

In [ ]:
path_save_figures = ''

In [ ]:
path_datasets = '/content/SeismicUP/Datasets_'

os.listdir(path_datasets)

['Cat_sis_SGC_TECTO_feb2014_2024',
 'Cat_sis_SGC_mar2018_2024',
 'Cat_mec_focal_tensor_mom_SGC_2014_2024',
 'README.md',
 'Cat_sis_SGC_jun1993_feb2018',
 'Cat_sis_SGC_LBG_jun1993_2021']

# Cargar catalogos

In [ ]:
lista_catalogos = ['Cat_sis_SGC_jun1993_feb2018',
                   'Cat_sis_SGC_mar2018_2024',
                   'Cat_sis_SGC_TECTO_feb2014_2024',
                   'Cat_sis_SGC_LBG_jun1993_2021']

for i in lista_catalogos:
  cantidad_archivos = len(os.listdir(os.path.join(path_datasets, i)))
  print(f'{i} = ', cantidad_archivos)

Cat_sis_SGC_jun1993_feb2018 =  26
Cat_sis_SGC_mar2018_2024 =  8
Cat_sis_SGC_TECTO_feb2014_2024 =  13
Cat_sis_SGC_LBG_jun1993_2021 =  29


In [ ]:
df_general = pd.DataFrame(columns=['Fecha-Hora UTC',
                                   'Latitud', # grados
                                   'Longitud', # grados
                                   'Profundidad [km]',
                                   'Magnitud',
                                   'Tipo Magnitud',
                                   'Error Latitud [km]',
                                   'Error Longitud [km]',
                                   'Error Profundidad [km]',
                                   'Numero de Fases',
                                   'RMS [seg]',
                                   'Gap', # grados
                                   'Departamento',
                                   'Municipio',
                                   ])

df_general.head()

,Fecha-Hora UTC,Latitud,Longitud,Profundidad [km],Magnitud,Tipo Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],Numero de Fases,RMS [seg],Gap,Departamento,Municipio


## Catalogo SGC 1

In [ ]:
path_catalogo = os.path.join(path_datasets, 'Cat_sis_SGC_jun1993_feb2018')
os.listdir(path_catalogo)

df_SGC_1 = df_general.copy()

for i in os.listdir(path_catalogo)[:]:
  if i.endswith('.xlsx'):
    print('-----'*3)
    print('File: ', i)
    print('-----'*3)
    df_temp_1 = sup.io.crear_catalogo_SGC_1(path_catalogo=path_catalogo,
                                            file_excel=i)

    df_SGC_1 = pd.concat([df_SGC_1, df_temp_1], ignore_index=True)
    df_SGC_1.reset_index(drop=True, inplace=True)

    del df_temp_1


---------------
File:  Cat_sis_SGC_2010.xlsx
---------------

['ANTIOQUIA' 'SANTANDER' 'CESAR' 'BOYACA' 'QUINDIO' 'VALLE DEL CAUCA'
 'TOLIMA' 'CUNDINAMARCA' 'CALDAS' 'CORDOBA' 'CASANARE' 'RISARALDA' 'META'
 'NORTE DE SANTANDER' 'BOLIVAR' 'ARAUCA' 'HUILA' 'SUCRE' 'LA GUAJIRA'
 'MAGDALENA']

['PUERTO BERRIO' 'SAN ANDRES' 'LOS SANTOS' 'CEPITA' 'SABANA DE TORRES'
 'CURUMANI' 'SABOYA' 'HATO' 'CALARCA' 'VALDIVIA' 'OBANDO' 'RIOBLANCO'
 'SAN PABLO DE BORBUR' 'CUCUNUBA' 'BARBOSA' 'VELEZ' 'EL CARMEN'
 'JERUSALEN' 'PALESTINA' 'BETULIA' 'ZAPATOCA' 'EL PLAYON'
 'SANTA HELENA DEL OPON' 'SUSA' 'PUERTO LIBERTADOR' 'OIBA' 'LA SALINA'
 'SAN VICENTE DE CHUCURI' 'CARTAGO' 'BALBOA' 'EL DORADO' 'ITUANGO'
 'CACHIRA' 'COPER' 'PUERTO PARRA' 'LA JAGUA DE IBIRICO' 'LEJANIAS'
 'CACERES' 'UTICA' 'ANORI' 'UBATE' 'LENGUAZAQUE' 'TIQUISIO' 'HERRAN'
 'GALAN' 'SURATA' 'MACANAL' 'MACEO' 'MUZO' 'CUBARA' 'BRICENO' 'TUNUNGUA'
 'NORCASIA' 'ALEJANDRIA' 'CHIMA' 'JESUS MARIA' 'MOGOTES' 'PISVA'
 'SAN PEDRO' 'ARENAL' 'EL PASO' 'G

/tmp/ipykernel_7303/915494741.py:14: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  df_SGC_1 = pd.concat([df_SGC_1, df_temp_1], ignore_index=True)



['SANTANDER' 'CUNDINAMARCA' 'BOYACA' 'META' 'ARAUCA' 'CASANARE' 'CORDOBA'
 'ANTIOQUIA' 'NORTE DE SANTANDER' 'SUCRE' 'QUINDIO' 'TOLIMA' 'HUILA'
 'CESAR' 'BOLIVAR' 'CALDAS' 'RISARALDA' 'VALLE DEL CAUCA' 'MAGDALENA']

['LOS SANTOS' 'PUERTO PARRA' 'SAN VICENTE DE CHUCURI' 'PALMAR'
 'VILLAPINZON' 'CERRITO' 'ALBANIA' 'SAN ANDRES' 'ZAPATOCA' 'CHIQUINQUIRA'
 'MALAGA' 'CONCEPCION' 'VILLANUEVA' 'CEPITA' 'GAMBITA' 'EL GUACAMAYO'
 'LA PAZ' 'URIBE' 'FORTUL' 'EL CARMEN' 'LA SALINA' 'SUCRE' 'OTANCHE'
 'CIENAGA DE ORO' 'PAIME' 'MONIQUIRA' 'PIEDECUESTA' 'COCORNA' 'YONDO'
 'LA ESPERANZA' 'EL TARRA' 'CALDAS' 'MIRAFLORES' 'VELEZ' 'ZETAQUIRA'
 'SABANA DE TORRES' 'SANTA HELENA DEL OPON' 'SANTA BARBARA' 'BETULIA'
 'SEGOVIA' 'ABREGO' 'CUCUNUBA' 'GUACA' 'LEBRIJA' 'BUENAVISTA' 'GALAN'
 'MESETAS' 'LA UNION' 'CONTRATACION' 'NECHI' 'BUCARASICA' 'CUBARA'
 'SARAVENA' 'GACHALA' 'TAME' 'GUICAN' 'ARATOCA' 'ARMENIA' 'ARAUCA'
 'JERUSALEN' 'TARAZA' 'PUERTO NARE' 'JORDAN' 'TAMARA'
 'SAN JOSE DE MIRANDA' 'SOGAMOSO' 'SURATA

In [ ]:
df_SGC_1['Catalogo'] = 'SGC 1'
df_SGC_1['Gap'] = df_SGC_1['Gap'].astype(float)
df_SGC_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145061 entries, 0 to 145060
Data columns (total 15 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Fecha-Hora UTC          145061 non-null  object 
 1   Latitud                 145061 non-null  float64
 2   Longitud                145061 non-null  float64
 3   Profundidad [km]        145061 non-null  float64
 4   Magnitud                145061 non-null  float64
 5   Tipo Magnitud           145061 non-null  object 
 6   Error Latitud [km]      144927 non-null  float64
 7   Error Longitud [km]     144927 non-null  float64
 8   Error Profundidad [km]  144927 non-null  float64
 9   Numero de Fases         145061 non-null  object 
 10  RMS [seg]               144927 non-null  float64
 11  Gap                     144927 non-null  float64
 12  Departamento            145061 non-null  object 
 13  Municipio               145061 non-null  object 
 14  Catalogo            

In [ ]:
df_SGC_1.head()

,Fecha-Hora UTC,Latitud,Longitud,Profundidad [km],Magnitud,Tipo Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],Numero de Fases,RMS [seg],Gap,Departamento,Municipio,Catalogo
0,2010-01-01 01:55:52,6.504,-74.471,4.0,1.6,Ml,10.0,4.2,11.9,5,0.6,220.0,ANTIOQUIA,PUERTO BERRIO,SGC 1
1,2010-01-01 04:19:45,6.775,-72.869,143.4,1.4,Ml,6.4,11.8,7.9,5,0.4,240.0,SANTANDER,SAN ANDRES,SGC 1
2,2010-01-01 07:33:57,6.773,-73.080,148.0,1.6,Ml,2.4,6.9,4.8,6,0.3,189.0,SANTANDER,LOS SANTOS,SGC 1
3,2010-01-01 07:40:27,6.776,-73.032,156.0,1.1,Ml,5.3,45.2,15.2,5,0.3,194.0,SANTANDER,CEPITA,SGC 1
4,2010-01-01 08:46:08,6.823,-73.023,137.6,1.9,Ml,5.4,10.2,11.4,9,0.5,135.0,SANTANDER,CEPITA,SGC 1


## Catalogo SGC 2

In [ ]:
path_catalogo = os.path.join(path_datasets, 'Cat_sis_SGC_mar2018_2024')
os.listdir(path_catalogo)

df_SGC_2 = df_general.copy()

for i in os.listdir(path_catalogo)[:]:
  if i.endswith('.xlsx'):
    print('-----'*3)
    print('File: ', i)
    print('-----'*3)
    df_temp_1 = sup.io.crear_catalogo_SGC_2(path_catalogo=path_catalogo,
                                            file_excel=i)

    df_SGC_2 = pd.concat([df_SGC_2, df_temp_1], ignore_index=True)
    df_SGC_2.reset_index(drop=True, inplace=True)

    del df_temp_1


---------------
File:  Cat_sis_SGC_2020.xlsx
---------------

['SANTANDER' 'META' 'RISARALDA' 'CUNDINAMARCA' 'NORTE DE SANTANDER'
 'BOLÍVAR' 'CESAR' 'ANTIOQUIA' 'TOLIMA' 'CÓRDOBA' 'VALLE DEL CAUCA'
 'BOYACÁ' 'MAGDALENA' 'CALDAS' 'QUINDÍO' 'VENEZUELA' 'LA GUAJIRA' 'HUILA'
 'CASANARE' 'NORTE COLOMBIA' 'ARAUCA' 'SUCRE' 'MAR CARIBE' 'ATLÁNTICO']

['RIONEGRO' 'LOS SANTOS' 'ZAPATOCA' 'URIBE' 'SANTA ROSA DE CABAL'
 'MESETAS' 'SAN VICENTE DE CHUCURÍ' 'CUCUNUBÁ' 'CHITAGÁ'
 'SAN MARTIN DE LOBA' 'YACOPÍ' 'PUERTO PARRA' 'AGUACHICA' 'EL CARMEN'
 'CIMITARRA' 'YONDÓ (CASABE)' 'LA PAZ' 'VILLANUEVA' 'BARICHARA' 'SIMACOTA'
 'FLORIÁN' 'CIUDAD BOLÍVAR' 'SAN MARTÍN' 'SAN BENITO' 'PLANADAS'
 'EL PLAYÓN' 'ACACÍAS' 'YOLOMBÓ' 'EL COPEY' 'AYAPEL' 'PURIFICACIÓN'
 'SABANA DE TORRES' 'PULÍ' 'LANDÁZURI' 'CHIRIGUANÁ' 'LA GLORIA'
 'LA ESPERANZA' 'SALGAR' 'CAICEDONIA' 'UBATÉ' 'SUCRE' 'RONCESVALLES'
 'SURATÁ' 'AGUSTÍN CODAZZI' 'GUAVATÁ' 'SANTUARIO' 'VÉLEZ' 'BOLIVAR'
 'PUERTO BOYACÁ' 'BETULIA' 'LEBRIJA' 'MARGARITA' 'VEN

/tmp/ipykernel_7303/1718883424.py:14: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  df_SGC_2 = pd.concat([df_SGC_2, df_temp_1], ignore_index=True)



['SANTANDER' 'ANTIOQUIA' 'BOLÍVAR' 'TOLIMA' 'CESAR' 'META' 'VENEZUELA'
 'NORTE DE SANTANDER' 'BOYACÁ' 'SUCRE' 'ARAUCA' 'CALDAS' 'NORTE COLOMBIA'
 'CUNDINAMARCA' 'VALLE DEL CAUCA' 'RISARALDA' 'CASANARE' 'MAGDALENA'
 'QUINDÍO' 'HUILA' 'LA GUAJIRA' 'CÓRDOBA' 'ATLÁNTICO' 'MAR CARIBE'
 'REGIÓN FRONTERIZA']

['LOS SANTOS' 'BOLIVAR' 'CAUCASIA' 'SUCRE' 'ACHÍ' 'VILLANUEVA' 'JORDÁN'
 'CAJAMARCA' 'CANTAGALLO' 'AGUACHICA' 'SABANA DE TORRES' 'IBAGUÉ' 'GIRÓN'
 'ACACÍAS' 'MESETAS' 'PUERTO GAITÁN' 'RIONEGRO' 'RIOVIEJO' nan 'ZAPATOCA'
 'SAN VICENTE DE CHUCURÍ' 'EL PLAYÓN' 'SIMACOTA' 'PUERTO WILCHES'
 'MURILLO' 'BETULIA' 'OCAÑA' 'BARICHARA' 'CALDAS' 'COPER' 'YONDÓ (CASABE)'
 'CUBARÁ' 'SAN MARCOS' 'EL CARMEN' 'MAGANGUÉ' 'VÉLEZ' 'SARAVENA' 'TOLEDO'
 'SANTA HELENA DEL OPÓN' 'VICTORIA' 'LA ESPERANZA' 'LENGUAZAQUE' 'TIBÚ'
 'CAÑASGORDAS' 'SAN MARTIN DE LOBA' 'TARAZÁ' 'VALPARAISO'
 'BARRANCABERMEJA' 'URIBE' 'SEVILLA' 'LEBRIJA' 'SAN MARTÍN' 'PUERTO PARRA'
 'ROVIRA' 'MISTRATÓ' 'CÁCHIRA' 'ANDES' 'PUEBLO BELLO' '

In [ ]:
df_SGC_2['Catalogo'] = 'SGC 2'
df_SGC_2['Gap'] = df_SGC_2['Gap'].astype(float)
df_SGC_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 134030 entries, 0 to 134029
Data columns (total 15 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Fecha-Hora UTC          134030 non-null  object 
 1   Latitud                 134030 non-null  float64
 2   Longitud                134030 non-null  float64
 3   Profundidad [km]        134030 non-null  float64
 4   Magnitud                134030 non-null  float64
 5   Tipo Magnitud           134030 non-null  object 
 6   Error Latitud [km]      134030 non-null  float64
 7   Error Longitud [km]     134030 non-null  float64
 8   Error Profundidad [km]  134030 non-null  float64
 9   Numero de Fases         134030 non-null  object 
 10  RMS [seg]               134030 non-null  float64
 11  Gap                     134030 non-null  float64
 12  Departamento            134030 non-null  object 
 13  Municipio               132275 non-null  object 
 14  Catalogo            

In [ ]:
df_SGC_2.head()

,Fecha-Hora UTC,Latitud,Longitud,Profundidad [km],Magnitud,Tipo Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],Numero de Fases,RMS [seg],Gap,Departamento,Municipio,Catalogo
0,2020-01-01 00:12:05,7.431,-73.228,132.82,1.4,MLr_vmm,1.344,1.344,1.8,21,0.3,106.0,SANTANDER,RIONEGRO,SGC 2
1,2020-01-01 00:13:16,6.792,-73.157,148.48,1.8,MLr_3,1.556,1.556,2.1,20,0.4,136.0,SANTANDER,LOS SANTOS,SGC 2
2,2020-01-01 00:14:37,6.848,-73.191,141.49,1.5,MLr_3,2.333,2.333,4.7,7,0.3,129.0,SANTANDER,ZAPATOCA,SGC 2
3,2020-01-01 00:16:17,6.829,-73.183,142.00,1.5,MLr_3,1.909,1.909,3.9,11,0.4,130.0,SANTANDER,ZAPATOCA,SGC 2
4,2020-01-01 00:40:18,3.136,-74.496,23.61,1.0,MLr_3,3.465,3.465,1.9,8,0.3,279.0,META,URIBE,SGC 2


## Catalogo TECTO

In [ ]:
path_catalogo = os.path.join(path_datasets, 'Cat_sis_SGC_TECTO_feb2014_2024')
os.listdir(path_catalogo)

df_SGC_TECTO = df_general.copy()

for i in os.listdir(path_catalogo)[:]:
  if i.endswith('.xlsx') and (i != 'excel_estaciones.xlsx'):
    print('-----'*3)
    print('File: ', i)
    print('-----'*3)
    df_temp_1 = sup.io.crear_catalogo_SGC_TECTO(path_catalogo=path_catalogo,
                                                file_excel=i)

    print(df_temp_1.shape)
    df_SGC_TECTO = pd.concat([df_SGC_TECTO, df_temp_1], ignore_index=True)
    df_SGC_TECTO.reset_index(drop=True, inplace=True)

    del df_temp_1


---------------
File:  Cat_sis_SGC_TECTO_2017.xlsx
---------------

['BOLÍVAR' 'ANTIOQUIA' 'NORTE DE SANTANDER' 'SANTANDER' 'MAGDALENA'
 'CESAR' 'BOYACÁ' 'ZULIA' 'CALDAS']

['NOROSÍ' 'PUERTO NARE' 'CÁCHIRA' 'LA PAZ' 'PUERTO WILCHES' 'RIONEGRO'
 'LA BELLEZA' 'ARIGUANÍ (EL DIFÍCIL)' 'TIQUISIO (PUERTO RICO)' 'PELAYA'
 'MONTECRISTO' 'BECERRIL' 'OTANCHE' 'AGUSTÍN CODAZZI' 'PUERTO PARRA'
 'PAILITAS' 'CIMITARRA' 'SABANA DE TORRES' 'LEBRIJA' 'YONDÓ (CASABE)'
 'EL CARMEN' 'LA JAGUA DE IBIRICO' 'SAN LUIS' 'MANAURE BALCÓN DEL CESAR'
 'PUERTO BERRÍO' 'CHIMICHAGUA' 'HATO' 'EL PLAYÓN' 'CURUMANÍ' 'EL BAGRE'
 'MACEO' 'EL PEÑÓN' 'REMEDIOS' 'SIMACOTA' 'SAN PABLO' 'SANTA ROSA DEL SUR'
 'EL PASO' 'ASTREA' 'BARRANCABERMEJA' 'ALGARROBO' 'CONVENCIÓN'
 'CHIRIGUANÁ' 'ZAPATOCA' 'GAMARRA' 'BETULIA' 'SIMITÍ' 'CANTAGALLO'
 'LANDÁZURI' 'CARACOLÍ' 'SAN MARTÍN' 'ALTOS DEL ROSARIO' 'AGUACHICA'
 'SABANAS DE SAN ÁNGEL' 'PUEBLO BELLO' 'SEGOVIA' 'PUERTO BOYACÁ'
 'BOSCONIA' 'EL COPEY' 'SANTA HELENA DEL OPÓN' 'MACHIQUES' 'O

/tmp/ipykernel_7303/343974462.py:15: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  df_SGC_TECTO = pd.concat([df_SGC_TECTO, df_temp_1], ignore_index=True)



['CESAR' 'SANTANDER' 'BOYACÁ' 'BOLÍVAR' 'NORTE DE SANTANDER' 'ANTIOQUIA'
 'MAGDALENA']

['LA GLORIA' 'CURUMANÍ' 'CIMITARRA' 'OTANCHE' 'LA PAZ' 'NOROSI' 'RIOVIEJO'
 'EL PASO' 'EL PEÑÓN' 'EL CARMEN' 'SAN PABLO' 'SABANA DE TORRES'
 'CANTAGALLO' 'ÁBREGO' 'LA BELLEZA' 'PUERTO BERRÍO' 'EL BANCO'
 'YONDÓ (CASABE)' 'CHIMICHAGUA' 'RIONEGRO' 'PAILITAS'
 'LA JAGUA DE IBIRICO' 'MORALES' 'REMEDIOS' 'SANTA HELENA DEL OPÓN'
 'SAN VICENTE DE CHUCURÍ' 'SAN ALBERTO' 'BARRANCABERMEJA' 'PUERTO WILCHES'
 'SAN FRANCISCO' 'BETULIA' 'PUERTO PARRA' 'ZAPATOCA' 'SONSÓN' 'CÁCHIRA'
 'LEBRIJA' 'LA ESPERANZA' 'PUERTO BOYACÁ' 'BOLIVAR' 'LANDÁZURI'
 'BECERRILL' 'PELAYA' 'SANTA ROSA DEL SUR' 'SIMITÍ' 'VEGACHÍ' 'SIMACOTA'
 'AGUSTÍN CODAZZI' 'CHIRIGUANÁ' 'GIRÓN' 'AGUACHICA' 'MACEO' 'TAMALAMEQUE'
 'VALLEDUPAR' 'SAN LUIS' 'SAN MARTIN DE LOBA' 'BOSCONIA'
 'ALTOS DEL ROSARIO' 'ARENAL' 'SUCRE' 'BARRANCO DE LOBA' 'YALÍ' 'CARACOLÍ'
 'GALÁN' 'GAMARRA' 'OCAÑA' 'PUERTO TRIUNFO' 'CONTRATACIÓN' 'YOLOMBÓ'
 'SAN MARTÍN' 'VÉLEZ' 'FLOR

In [ ]:
df_SGC_TECTO['Catalogo'] = 'TECTO'
df_SGC_TECTO['Profundidad [km]'] = df_SGC_TECTO['Profundidad [km]'].astype(float)
df_SGC_TECTO['Gap'] = df_SGC_TECTO['Gap'].astype(float)
df_SGC_TECTO.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7668 entries, 0 to 7667
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Fecha-Hora UTC          7668 non-null   object 
 1   Latitud                 7668 non-null   float64
 2   Longitud                7668 non-null   float64
 3   Profundidad [km]        7668 non-null   float64
 4   Magnitud                7668 non-null   float64
 5   Tipo Magnitud           7668 non-null   object 
 6   Error Latitud [km]      7668 non-null   float64
 7   Error Longitud [km]     7668 non-null   float64
 8   Error Profundidad [km]  7668 non-null   float64
 9   Numero de Fases         0 non-null      float64
 10  RMS [seg]               7668 non-null   float64
 11  Gap                     7668 non-null   float64
 12  Departamento            7668 non-null   object 
 13  Municipio               7668 non-null   object 
 14  Catalogo                7668 non-null   

In [ ]:
df_SGC_TECTO.head()

,Fecha-Hora UTC,Latitud,Longitud,Profundidad [km],Magnitud,Tipo Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],Numero de Fases,RMS [seg],Gap,Departamento,Municipio,Catalogo
0,2017-01-01 02:37:44,8.574,-73.969,19.6,0.8,MLr_vmm,1.5,2.5,3.3,NaN,0.1,188.0,BOLÍVAR,NOROSÍ,TECTO
1,2017-01-01 03:49:24,6.306,-74.647,12.7,1.0,MLr_vmm,4.6,4.1,5.3,NaN,0.3,271.0,ANTIOQUIA,PUERTO NARE,TECTO
2,2017-01-01 09:08:39,7.648,-73.045,0.0,0.7,MLr_vmm,0.8,3.1,3.7,NaN,0.1,218.0,NORTE DE SANTANDER,CÁCHIRA,TECTO
3,2017-01-01 20:54:20,6.204,-73.698,32.0,0.9,MLr_vmm,2.3,1.7,8.3,NaN,0.2,177.0,SANTANDER,LA PAZ,TECTO
4,2017-01-02 10:01:10,7.291,-73.900,33.0,1.3,MLr_vmm,3.9,2.1,6.6,NaN,0.3,162.0,SANTANDER,PUERTO WILCHES,TECTO


## Catalogo LBG

In [ ]:
path_catalogo = os.path.join(path_datasets, 'Cat_sis_SGC_LBG_jun1993_2021')
os.listdir(path_catalogo)

df_SGC_LBG = df_general.copy()

for i in os.listdir(path_catalogo)[:]:
  if i.endswith('.xlsx'):
    print('-----'*3)
    print('File: ', i)
    print('-----'*3)
    df_temp_1 = sup.io.crear_catalogo_SGC_LBG(path_catalogo=path_catalogo,
                                              file_excel=i)

    print(df_temp_1.shape)
    df_SGC_LBG = pd.concat([df_SGC_LBG, df_temp_1], ignore_index=True)
    df_SGC_LBG.reset_index(drop=True, inplace=True)

    del df_temp_1


---------------
File:  Cat_sis_SGC_LBG_1994.xlsx
---------------

[nan]

[nan]

Index(['Fecha  (UTC)', 'Hora  (UTC)', 'Latitud(°)', 'Longitud(°)',
       'Profundidad(Km)', 'Magnitud', 'Tipo Magnitud', 'Rms(Seg)', 'Gap(°)',
       'Error  Latitud(Km)', 'Error  Longitud(Km)', 'Error  Profundidad(Km)',
       'Modelo', 'Fecha-Hora UTC', 'Departamento', 'Municipio', 'FASES'],
      dtype='object')
Eventos:  228
(228, 14)
---------------
File:  Cat_sis_SGC_LBG_1997.xlsx
---------------

[nan]

[nan]

Index(['Fecha  (UTC)', 'Hora  (UTC)', 'Latitud(°)', 'Longitud(°)',
       'Profundidad(Km)', 'Magnitud', 'Tipo Magnitud', 'Rms(Seg)', 'Gap(°)',
       'Error  Latitud(Km)', 'Error  Longitud(Km)', 'Error  Profundidad(Km)',
       'Modelo', 'Fecha-Hora UTC', 'Departamento', 'Municipio', 'FASES'],
      dtype='object')
Eventos:  129
(129, 14)
---------------
File:  Cat_sis_SGC_LBG_2020.xlsx
---------------

[nan]

[nan]

Index(['Fecha  (UTC)', 'Hora  (UTC)', 'Latitud(°)', 'Longitud(°)',
       'P

/tmp/ipykernel_7303/2212349628.py:15: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  df_SGC_LBG = pd.concat([df_SGC_LBG, df_temp_1], ignore_index=True)



[nan]

[nan]

Index(['Fecha  (UTC)', 'Hora  (UTC)', 'Latitud(°)', 'Longitud(°)',
       'Profundidad(Km)', 'Magnitud', 'Tipo Magnitud', 'Rms(Seg)', 'Gap(°)',
       'Error  Latitud(Km)', 'Error  Longitud(Km)', 'Error  Profundidad(Km)',
       'Modelo', 'Fecha-Hora UTC', 'Departamento', 'Municipio', 'FASES'],
      dtype='object')
Eventos:  1091
(1091, 14)
---------------
File:  Cat_sis_SGC_LBG_1999.xlsx
---------------

[nan]

[nan]

Index(['Fecha  (UTC)', 'Hora  (UTC)', 'Latitud(°)', 'Longitud(°)',
       'Profundidad(Km)', 'Magnitud', 'Tipo Magnitud', 'Rms(Seg)', 'Gap(°)',
       'Error  Latitud(Km)', 'Error  Longitud(Km)', 'Error  Profundidad(Km)',
       'Modelo', 'Fecha-Hora UTC', 'Departamento', 'Municipio', 'FASES'],
      dtype='object')
Eventos:  200
(200, 14)
---------------
File:  Cat_sis_SGC_LBG_2016.xlsx
---------------

[nan]

[nan]

Index(['Fecha  (UTC)', 'Hora  (UTC)', 'Latitud(°)', 'Longitud(°)',
       'Profundidad(Km)', 'Magnitud', 'Tipo Magnitud', 'Rms(Seg)', 'Gap(

In [ ]:
df_SGC_LBG['Catalogo'] = 'LBG'
df_SGC_LBG.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16481 entries, 0 to 16480
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Fecha-Hora UTC          16481 non-null  object 
 1   Latitud                 16481 non-null  float64
 2   Longitud                16481 non-null  float64
 3   Profundidad [km]        16481 non-null  float64
 4   Magnitud                16481 non-null  float64
 5   Tipo Magnitud           16481 non-null  object 
 6   Error Latitud [km]      16481 non-null  float64
 7   Error Longitud [km]     16481 non-null  float64
 8   Error Profundidad [km]  16481 non-null  float64
 9   Numero de Fases         0 non-null      float64
 10  RMS [seg]               16481 non-null  float64
 11  Gap                     16481 non-null  object 
 12  Departamento            0 non-null      object 
 13  Municipio               0 non-null      object 
 14  Catalogo                16481 non-null

In [ ]:
df_SGC_LBG.head()

,Fecha-Hora UTC,Latitud,Longitud,Profundidad [km],Magnitud,Tipo Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],Numero de Fases,RMS [seg],Gap,Departamento,Municipio,Catalogo
0,1994-01-01 13:20:05,4.936,-74.605,11.6,1.8,ML,2.2,2.0,15.1,NaN,0.4,128,NaN,NaN,LBG
1,1994-01-01 03:52:24,4.828,-74.563,17.3,1.8,ML,9.0,10.8,11.0,NaN,0.7,127,NaN,NaN,LBG
2,1994-01-01 15:24:32,4.739,-74.682,8.1,1.8,ML,10.6,10.3,25.9,NaN,0.5,102,NaN,NaN,LBG
3,1994-01-01 20:10:36,5.702,-74.598,3.4,2.5,ML,3.6,3.6,9.3,NaN,1.0,116,NaN,NaN,LBG
4,1994-01-03 07:37:51,5.231,-74.730,14.0,1.7,ML,2.3,2.6,6.8,NaN,0.4,144,NaN,NaN,LBG


# Unir catalogos

In [ ]:
df_unido = pd.concat([df_SGC_1, df_SGC_2, df_SGC_TECTO, df_SGC_LBG], ignore_index=True)
df_unido.reset_index(drop=True, inplace=True)

df_unido['Tiempo'] = pd.to_datetime(df_unido['Fecha-Hora UTC'])

df_unido.head()

,Fecha-Hora UTC,Latitud,Longitud,Profundidad [km],Magnitud,Tipo Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],Numero de Fases,RMS [seg],Gap,Departamento,Municipio,Catalogo,Tiempo
0,2010-01-01 01:55:52,6.504,-74.471,4.0,1.6,Ml,10.0,4.2,11.9,5,0.6,220.0,ANTIOQUIA,PUERTO BERRIO,SGC 1,2010-01-01 01:55:52
1,2010-01-01 04:19:45,6.775,-72.869,143.4,1.4,Ml,6.4,11.8,7.9,5,0.4,240.0,SANTANDER,SAN ANDRES,SGC 1,2010-01-01 04:19:45
2,2010-01-01 07:33:57,6.773,-73.080,148.0,1.6,Ml,2.4,6.9,4.8,6,0.3,189.0,SANTANDER,LOS SANTOS,SGC 1,2010-01-01 07:33:57
3,2010-01-01 07:40:27,6.776,-73.032,156.0,1.1,Ml,5.3,45.2,15.2,5,0.3,194.0,SANTANDER,CEPITA,SGC 1,2010-01-01 07:40:27
4,2010-01-01 08:46:08,6.823,-73.023,137.6,1.9,Ml,5.4,10.2,11.4,9,0.5,135.0,SANTANDER,CEPITA,SGC 1,2010-01-01 08:46:08


In [ ]:
df_unido['Catalogo'].unique()

array(['SGC 1', 'SGC 2', 'TECTO', 'LBG'], dtype=object)

In [ ]:
df_unido.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303240 entries, 0 to 303239
Data columns (total 16 columns):
 #   Column                  Non-Null Count   Dtype         
---  ------                  --------------   -----         
 0   Fecha-Hora UTC          303240 non-null  object        
 1   Latitud                 303240 non-null  float64       
 2   Longitud                303240 non-null  float64       
 3   Profundidad [km]        303240 non-null  float64       
 4   Magnitud                303240 non-null  float64       
 5   Tipo Magnitud           303240 non-null  object        
 6   Error Latitud [km]      303106 non-null  float64       
 7   Error Longitud [km]     303106 non-null  float64       
 8   Error Profundidad [km]  303106 non-null  float64       
 9   Numero de Fases         279091 non-null  object        
 10  RMS [seg]               303106 non-null  float64       
 11  Gap                     303106 non-null  object        
 12  Departamento            286759

In [ ]:
df_unido.describe().round(2)

,Latitud,Longitud,Profundidad [km],Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],RMS [seg],Tiempo
count,303240.00,303240.00,303240.00,303240.00,303106.00,303106.00,303106.00,303106.00,303240
mean,6.58,-73.51,101.06,1.90,4.46,5.95,7.69,0.49,2015-07-06 19:47:00.099531520
min,3.00,-76.00,-4.97,0.00,0.00,0.00,0.00,0.00,1993-06-01 01:32:55
25%,6.42,-73.77,30.13,1.50,2.00,2.10,3.40,0.30,2011-08-20 23:17:01.750000128
50%,6.79,-73.18,136.10,1.80,3.40,4.07,6.00,0.40,2017-08-08 04:09:50.500000
75%,6.84,-73.12,145.60,2.20,5.30,7.00,8.70,0.60,2021-04-17 14:51:00.750000128
max,11.00,-71.01,594.80,6.80,999.00,999.00,999.00,10.00,2024-12-31 23:39:56
std,1.23,0.73,58.40,0.66,8.58,8.83,19.56,0.28,NaN


## Duplicados

In [ ]:
df_unido.duplicated(subset=['Fecha-Hora UTC', 'Latitud', 'Longitud'], keep='first').sum()
#  duplicados

np.int64(22154)

In [ ]:
df_unido['Tipo Magnitud'].unique()

array(['Ml', 'Mw', 'MLr_vmm', 'MLr_3', 'MLr_2', 'MLr_1', 'MLr_4',
       'Mw(mB)', 'M_MLr', 'MLr_PtoGtn', 'MLr', 'M', 'MLr_5', 'mb', 'MLv',
       'mB', 'M_Pac', 'Mwp', 'ML'], dtype=object)

In [ ]:
df_unido['Tipo Magnitud 2'] = df_unido['Tipo Magnitud'].str.lower()
df_unido['Tipo Magnitud 2'].unique()

array(['ml', 'mw', 'mlr_vmm', 'mlr_3', 'mlr_2', 'mlr_1', 'mlr_4',
       'mw(mb)', 'm_mlr', 'mlr_ptogtn', 'mlr', 'm', 'mlr_5', 'mb', 'mlv',
       'm_pac', 'mwp'], dtype=object)

In [ ]:
df_unido.describe()

,Latitud,Longitud,Profundidad [km],Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],RMS [seg],Tiempo
count,303240.000000,303240.000000,303240.000000,303240.000000,303106.000000,303106.000000,303106.000000,303106.000000,303240
mean,6.580466,-73.511873,101.059301,1.904443,4.462535,5.949131,7.689872,0.492964,2015-07-06 19:47:00.099531520
min,3.000000,-76.000000,-4.970000,0.000000,0.000000,0.000000,0.000000,0.000000,1993-06-01 01:32:55
25%,6.416000,-73.768000,30.127500,1.500000,2.000000,2.100000,3.400000,0.300000,2011-08-20 23:17:01.750000128
50%,6.787000,-73.181500,136.100000,1.800000,3.400000,4.069000,6.000000,0.400000,2017-08-08 04:09:50.500000
75%,6.836000,-73.122000,145.600000,2.200000,5.303000,7.002750,8.700000,0.600000,2021-04-17 14:51:00.750000128
max,11.000000,-71.011000,594.800000,6.800000,999.000000,999.000000,999.000000,10.000000,2024-12-31 23:39:56
std,1.231248,0.728092,58.400599,0.664271,8.581627,8.831882,19.562266,0.275579,NaN


In [ ]:
df_unido.sort_values(by=['Tiempo', 'Tipo Magnitud 2'],
                     ascending=True,
                     inplace=True)

Se considerarán únicamente los eventos que cuenten con ML como tipo de magnitud.

In [ ]:
filtro_ml = df_unido['Tipo Magnitud 2'].str.contains('ml')

df_unido[filtro_ml]['Tipo Magnitud 2'].unique()

array(['ml', 'mlr', 'mlr_vmm', 'mlr_4', 'mlr_2', 'mlr_3', 'm_mlr',
       'mlr_1', 'mlr_ptogtn', 'mlr_5', 'mlv'], dtype=object)

In [ ]:
df_unido.duplicated(subset=['Fecha-Hora UTC', 'Latitud', 'Longitud'], keep='first').sum()
# 22154 duplicados

np.int64(22154)

In [ ]:
df_filtrado = df_unido.copy()
df_filtrado.drop_duplicates(subset=['Fecha-Hora UTC', 'Latitud', 'Longitud'], keep='first', inplace=True)
df_filtrado.reset_index(drop=True, inplace=True)

In [ ]:
df_filtrado.duplicated(subset=['Fecha-Hora UTC', 'Latitud', 'Longitud'], keep='first').sum()
# Ya no se tienen duplicados

np.int64(0)

In [ ]:
df_filtrado.describe().round(2)

,Latitud,Longitud,Profundidad [km],Magnitud,Error Latitud [km],Error Longitud [km],Error Profundidad [km],RMS [seg],Tiempo
count,281086.00,281086.00,281086.00,281086.00,280952.00,280952.00,280952.00,280952.00,281086
mean,6.58,-73.47,106.61,1.91,4.46,6.05,7.35,0.50,2015-08-06 12:41:22.378137344
min,3.00,-76.00,-4.97,0.00,0.00,0.00,0.00,0.00,1993-06-01 01:32:55
25%,6.54,-73.70,62.00,1.50,2.09,2.18,3.40,0.30,2011-07-24 20:23:13.249999872
50%,6.79,-73.18,138.00,1.80,3.50,4.20,6.10,0.40,2017-10-14 19:12:13.500000
75%,6.83,-73.12,146.00,2.20,5.40,7.21,8.70,0.60,2021-06-27 06:10:59.750000128
max,11.00,-71.01,594.80,6.80,313.47,283.69,832.90,10.00,2024-12-31 23:39:56
std,1.21,0.72,55.74,0.64,4.88,7.01,8.22,0.28,NaN


In [ ]:
df_filtrado[df_filtrado['Profundidad [km]'] <= 60].to_csv('df_lq60km.csv',
                                                          decimal=',',
                                                          sep=';',
                                                          index=False)

## Geodataframe

In [ ]:
gdf_filtrado = gpd.GeoDataFrame(
    df_filtrado, geometry=gpd.points_from_xy(df_filtrado.Longitud, df_filtrado.Latitud), crs="EPSG:4326"
)

In [ ]:
gdf_filtrado = gdf_filtrado.to_crs(9377)

In [ ]:
df_filtrado['X [m]'] = gdf_filtrado.get_coordinates()['x']
df_filtrado['Y [m]'] = gdf_filtrado.get_coordinates()['y']

df_filtrado['X [km]'] = df_filtrado['X [m]'] / 1000
df_filtrado['Y [km]'] = df_filtrado['Y [m]'] / 1000

df_filtrado.head()

## Punto referencia Campo Colorado



In [ ]:
punto_xy_CampoColorado = (4919222.973, 2308553.476) # x, y en metros

# se va calcular la distancia de los eventos al centroide o punto de referencia
# del Campo Colorado solo en XY mas no en profundidad

df_filtrado['Distancia [m]'] = np.sqrt((df_filtrado['X [m]'] - punto_xy_CampoColorado[0])**2 + (df_filtrado['Y [m]'] - punto_xy_CampoColorado[1])**2)
df_filtrado['Distancia [km]'] = df_filtrado['Distancia [m]'] / 1000

df_filtrado.describe().round(2)

## Subcatalogos

In [ ]:
limites_BSN = {'longitud': [-73.40, -72.80],
               'latitud': [6.50, 7.10]}
limites_CC = {'longitud': [-74.19, -73.26],
              'latitud': [6.32, 7.25]}


In [ ]:
df_FC = df_filtrado.copy()

filtro_BSN = (df_filtrado['Longitud'] >= limites_BSN['longitud'][0]) & (df_filtrado['Longitud'] <= limites_BSN['longitud'][1]) & (df_filtrado['Latitud'] >= limites_BSN['latitud'][0]) & (df_filtrado['Latitud'] <= limites_BSN['latitud'][1])
df_BSN = df_filtrado[filtro_BSN].copy()

filtro_CC = (df_filtrado['Longitud'] >= limites_CC['longitud'][0]) & (df_filtrado['Longitud'] <= limites_CC['longitud'][1]) & (df_filtrado['Latitud'] >= limites_CC['latitud'][0]) & (df_filtrado['Latitud'] <= limites_CC['latitud'][1])
df_CC = df_filtrado[filtro_CC].copy()


In [ ]:
df_FC.describe().round(2)

In [ ]:
df_BSN.describe().round(2)

In [ ]:
df_CC.describe().round(2)

In [ ]:
total_eventos_FC = df_FC.shape[0]
total_eventos_BSN = df_BSN.shape[0]
total_eventos_CC = df_CC.shape[0]

porcentaje_eventos_FC = (total_eventos_FC / total_eventos_FC) * 100
porcentaje_eventos_BSN = (total_eventos_BSN / total_eventos_FC) * 100
porcentaje_eventos_CC = (total_eventos_CC / total_eventos_FC) * 100

print(f'Total de eventos FC: {total_eventos_FC} [{porcentaje_eventos_FC:.2f}%]')
print(f'Total de eventos BSN: {total_eventos_BSN} [{porcentaje_eventos_BSN:.2f}%]')
print(f'Total de eventos CC: {total_eventos_CC} [{porcentaje_eventos_CC:.2f}%]')

Después de aplicar el filtrado del catálogo (considerando únicamente eventos con ML como tipo de magnitud), se obtiene:

* `FC` (Full Catalog): 281086 eventos (100% del conjunto de estudio).
* `BSN` (Bucaramanga Seismic Nest Catalog): 161158 eventos, equivalentes al 57.33% del `FC`.
* `CC` (Campo Colorado Catalog): 22052 eventos, equivalentes al 7.85% del `FC`.


# Histogramas errores

In [ ]:
fig, ax = plt.subplots(nrows=2, ncols=3, figsize=(12,8))

# FC
sup.plots.histograma_plot(df= df_FC,
                          columna= 'X [km]',
                          ax_h= ax[0,0],
                          dist_min= df_FC['X [km]'].min(),
                          units_label='km',
                          step_= 20)
sup.plots.histograma_plot(df= df_FC,
                          columna= 'Error Longitud [km]',
                          ax_h= ax[1,0],
                          dist_min= 0,
                          units_label='km',
                          step_= 10)
ax[0,0].set_xlabel('X [km] - FC')
ax[1,0].set_xlabel('X Error [km] - FC')

# BSN
sup.plots.histograma_plot(df= df_BSN,
                          columna= 'X [km]',
                          ax_h= ax[0,1],
                          dist_min= df_BSN['X [km]'].min(),
                          units_label='km',
                          step_= 5)
sup.plots.histograma_plot(df= df_BSN,
                          columna= 'Error Longitud [km]',
                          ax_h= ax[1,1],
                          dist_min= 0,
                          units_label='km',
                          step_= 5)
ax[0,1].set_xlabel('X [km] - BSN')
ax[1,1].set_xlabel('X Error [km] - BSN')

# CC
sup.plots.histograma_plot(df= df_CC,
                          columna= 'X [km]',
                          ax_h= ax[0,2],
                          dist_min= df_CC['X [km]'].min(),
                          units_label='km',
                          step_= 5)
sup.plots.histograma_plot(df= df_CC,
                          columna= 'Error Longitud [km]',
                          ax_h= ax[1,2],
                          dist_min= 0,
                          units_label='km',
                          step_= 5)
ax[0,2].set_xlabel('X [km] - CC')
ax[1,2].set_xlabel('X Error [km] - CC')

ax[0,0].set_title('FC Catalog')
ax[0,1].set_title('BSN Catalog')
ax[0,2].set_title('CC Catalog')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'hits_errores_coorX',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

In [ ]:
fig, ax = plt.subplots(nrows=2, ncols=3, figsize=(12,8))

# FC
sup.plots.histograma_plot(df= df_FC,
                          columna= 'Y [km]',
                          ax_h= ax[0,0],
                          dist_min= df_FC['Y [km]'].min(),
                          step_= 20)
sup.plots.histograma_plot(df= df_FC,
                          columna= 'Error Latitud [km]',
                          ax_h= ax[1,0],
                          dist_min= 0,
                          step_= 10)
ax[0,0].set_xlabel('Y [km] - FC')
ax[1,0].set_xlabel('Y Error [km] - FC')

# BSN
sup.plots.histograma_plot(df= df_BSN,
                          columna= 'Y [km]',
                          ax_h= ax[0,1],
                          dist_min= df_BSN['Y [km]'].min(),
                          step_= 5)
sup.plots.histograma_plot(df= df_BSN,
                          columna= 'Error Latitud [km]',
                          ax_h= ax[1,1],
                          dist_min= 0,
                          step_= 5)
ax[0,1].set_xlabel('Y [km] - BSN')
ax[1,1].set_xlabel('Y Error [km] - BSN')

# CC
sup.plots.histograma_plot(df= df_CC,
                          columna= 'Y [km]',
                          ax_h= ax[0,2],
                          dist_min= df_CC['Y [km]'].min(),
                          step_= 5)
sup.plots.histograma_plot(df= df_CC,
                          columna= 'Error Latitud [km]',
                          ax_h= ax[1,2],
                          dist_min= 0,
                          step_= 5)
ax[0,2].set_xlabel('Y [km] - CC')
ax[1,2].set_xlabel('Y Error [km] - CC')

ax[0,0].set_title('FC Catalog')
ax[0,1].set_title('BSN Catalog')
ax[0,2].set_title('CC Catalog')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'hits_errores_coorY',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

In [ ]:
fig, ax = plt.subplots(nrows=2, ncols=3, figsize=(12,8))

# FC
sup.plots.histograma_plot(df= df_FC,
                          columna= 'Profundidad [km]',
                          ax_h= ax[0,0],
                          dist_min= df_FC['Profundidad [km]'].min(),
                          step_= 20)
sup.plots.histograma_plot(df= df_FC,
                          columna= 'Error Profundidad [km]',
                          ax_h= ax[1,0],
                          dist_min= 0,
                          step_= 20)
ax[0,0].set_xlabel('Depth [km] - FC')
ax[1,0].set_xlabel('Depth Error [km] - FC')

# BSN
sup.plots.histograma_plot(df= df_BSN,
                          columna= 'Profundidad [km]',
                          ax_h= ax[0,1],
                          dist_min= df_BSN['Profundidad [km]'].min(),
                          step_= 20)
sup.plots.histograma_plot(df= df_BSN,
                          columna= 'Error Profundidad [km]',
                          ax_h= ax[1,1],
                          dist_min= 0,
                          step_= 5)
ax[0,1].set_xlabel('Depth [km] - BSN')
ax[1,1].set_xlabel('Depth Error [km] - BSN')

# CC
sup.plots.histograma_plot(df= df_CC,
                          columna= 'Profundidad [km]',
                          ax_h= ax[0,2],
                          dist_min= df_CC['Profundidad [km]'].min(),
                          step_= 10)
sup.plots.histograma_plot(df= df_CC,
                          columna= 'Error Profundidad [km]',
                          ax_h= ax[1,2],
                          dist_min= 0,
                          step_= 5)
ax[0,2].set_xlabel('Depth [km] - CC')
ax[1,2].set_xlabel('Depth Error [km] - CC')

ax[0,0].set_title('FC Catalog')
ax[0,1].set_title('BSN Catalog')
ax[0,2].set_title('CC Catalog')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'hits_errores_profundidad',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

# Tiempo vs magnitud

## Profundidad limite

In [ ]:
profundidad_corte = 60

In [ ]:
fig, ax = plt.subplots(ncols=1, nrows=3, figsize=(8,11))

# FC
filtro_datos = df_FC['Profundidad [km]'] <= profundidad_corte
sup.plots.time_magnitud_plot(df = df_FC,
                             filtro_datos = filtro_datos,
                             columna_eje_x_tiempo = 'Tiempo',
                             comlumna_eje_y = 'Magnitud',
                             ax_s = ax[0],
                             label_s= f'{filtro_datos.sum()} Earthquakes [Depth $\leq{profundidad_corte}$ km]')
ax[0].legend()
ax[0].set_xlabel('Time')
ax[0].set_ylabel('Magnitude')
ax[0].set_title('FC Catalog')


# BSN
filtro_datos = df_BSN['Profundidad [km]'] <= profundidad_corte
sup.plots.time_magnitud_plot(df = df_BSN,
                             filtro_datos = filtro_datos,
                             columna_eje_x_tiempo = 'Tiempo',
                             comlumna_eje_y = 'Magnitud',
                             ax_s = ax[1],
                             label_s= f'{filtro_datos.sum()} Earthquakes [Depth $\leq{profundidad_corte}$ km]')
ax[1].legend()
ax[1].set_xlabel('Time')
ax[1].set_ylabel('Magnitude')
ax[1].set_title('BSN Catalog')

# CC
filtro_datos = df_CC['Profundidad [km]'] <= profundidad_corte
sup.plots.time_magnitud_plot(df = df_CC,
                             filtro_datos = filtro_datos,
                             columna_eje_x_tiempo = 'Tiempo',
                             comlumna_eje_y = 'Magnitud',
                             ax_s = ax[2],
                             label_s= f'{filtro_datos.sum()} Earthquakes [Depth $\leq{profundidad_corte}$ km]')
ax[2].legend()
ax[2].set_xlabel('Time')
ax[2].set_ylabel('Magnitude')
ax[2].set_title('CC Catalog')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'scatter_tiempo_magnitud_somero',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

In [ ]:
fig, ax = plt.subplots(ncols=1, nrows=3, figsize=(8,11))

# FC
filtro_datos = df_FC['Profundidad [km]'] > profundidad_corte
sup.plots.time_magnitud_plot(df = df_FC,
                             filtro_datos = filtro_datos,
                             columna_eje_x_tiempo = 'Tiempo',
                             comlumna_eje_y = 'Magnitud',
                             ax_s = ax[0],
                             label_s= f'{filtro_datos.sum()} Earthquakes [Depth $>{profundidad_corte}$ km]')
ax[0].legend()
ax[0].set_xlabel('Time')
ax[0].set_ylabel('Magnitude')
ax[0].set_title('FC Catalog')


# BSN
filtro_datos = df_BSN['Profundidad [km]'] > profundidad_corte
sup.plots.time_magnitud_plot(df = df_BSN,
                             filtro_datos = filtro_datos,
                             columna_eje_x_tiempo = 'Tiempo',
                             comlumna_eje_y = 'Magnitud',
                             ax_s = ax[1],
                             label_s= f'{filtro_datos.sum()} Earthquakes [Depth $>{profundidad_corte}$ km]')
ax[1].legend()
ax[1].set_xlabel('Time')
ax[1].set_ylabel('Magnitude')
ax[1].set_title('BSN Catalog')

# CC
filtro_datos = df_CC['Profundidad [km]'] > profundidad_corte
sup.plots.time_magnitud_plot(df = df_CC,
                             filtro_datos = filtro_datos,
                             columna_eje_x_tiempo = 'Tiempo',
                             comlumna_eje_y = 'Magnitud',
                             ax_s = ax[2],
                             label_s= f'{filtro_datos.sum()} Earthquakes [Depth $>{profundidad_corte}$ km]')
ax[2].legend()
ax[2].set_xlabel('Time')
ax[2].set_ylabel('Magnitude')
ax[2].set_title('CC Catalog')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'scatter_tiempo_magnitud_profundos',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

# Histograma magnitudes

In [ ]:
steps_magnitud = 0.2
mag_min = 0
mag_max = 8

fig, ax = plt.subplots(nrows=2, ncols=3, figsize=(12,8))

# FC
# <= profunidad corte
filtro_datos = df_FC['Profundidad [km]'] <= profundidad_corte
df_FC_temp = df_FC[filtro_datos]
sup.plots.histograma_plot(df= df_FC_temp,
                          columna= 'Magnitud',
                          ax_h= ax[0,0],
                          dist_min= mag_min,
                          dist_max= mag_max,
                          units_label='Ml',
                          step_= steps_magnitud)
ax[0,0].set_xlabel(f'Magnitude [Depth $\leq{profundidad_corte}$ km] - FC')
# > profunidad corte
filtro_datos = df_FC['Profundidad [km]'] > profundidad_corte
df_FC_temp = df_FC[filtro_datos]
sup.plots.histograma_plot(df= df_FC_temp,
                          columna= 'Magnitud',
                          ax_h= ax[1,0],
                          dist_min= mag_min,
                          dist_max= mag_max,
                          units_label='Ml',
                          step_= steps_magnitud)
ax[1,0].set_xlabel(f'Magnitude [Depth $>{profundidad_corte}$ km] - FC')

# BSN
# <= profunidad corte
filtro_datos = df_BSN['Profundidad [km]'] <= profundidad_corte
df_BSN_temp = df_BSN[filtro_datos]
sup.plots.histograma_plot(df= df_BSN_temp,
                          columna= 'Magnitud',
                          ax_h= ax[0,1],
                          dist_min= mag_min,
                          dist_max= mag_max,
                          units_label='Ml',
                          step_= steps_magnitud)
ax[0,1].set_xlabel(f'Magnitude [Depth $\leq{profundidad_corte}$ km] - BSN')
# > profunidad corte
filtro_datos = df_BSN['Profundidad [km]'] > profundidad_corte
df_BSN_temp = df_BSN[filtro_datos]
sup.plots.histograma_plot(df= df_BSN_temp,
                          columna= 'Magnitud',
                          ax_h= ax[1,1],
                          dist_min= mag_min,
                          dist_max= mag_max,
                          units_label='Ml',
                          step_= steps_magnitud)
ax[1,1].set_xlabel(f'Magnitude [Depth $>{profundidad_corte}$ km] - BSN')

# CC
# <= profunidad corte
filtro_datos = df_CC['Profundidad [km]'] <= profundidad_corte
df_CC_temp = df_CC[filtro_datos]
sup.plots.histograma_plot(df= df_CC_temp,
                          columna= 'Magnitud',
                          ax_h= ax[0,2],
                          dist_min= mag_min,
                          dist_max= mag_max,
                          units_label='Ml',
                          step_= steps_magnitud)
ax[0,2].set_xlabel(f'Magnitude [Depth $\leq{profundidad_corte}$ km] - CC')
# > profunidad corte
filtro_datos = df_CC['Profundidad [km]'] > profundidad_corte
df_CC_temp = df_CC[filtro_datos]
sup.plots.histograma_plot(df= df_CC_temp,
                          columna= 'Magnitud',
                          ax_h= ax[1,2],
                          dist_min= mag_min,
                          dist_max= mag_max,
                          units_label='Ml',
                          step_= steps_magnitud)
ax[1,2].set_xlabel(f'Magnitude [Depth $>{profundidad_corte}$ km] - CC')

ax[0,0].set_title('FC Catalog')
ax[0,1].set_title('BSN Catalog')
ax[0,2].set_title('CC Catalog')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'hits_magnitudes_',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

## Cantidad eventos al año

In [ ]:
year_range = np.arange(df_FC['Tiempo'].dt.year.min(),
                       df_FC['Tiempo'].dt.year.max()+1)

fig, ax = plt.subplots(ncols=1, nrows=3, figsize=(7,9), sharex=True)

# Todos
sns.histplot(x = df_FC['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[0],
             legend= True,
             label= f'FC Catalog',
             element='step',
             fill=False)
sns.histplot(x = df_BSN['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[0],
             legend= True,
             label= f'BSN Catalog',
             element='step',
             fill=False)
sns.histplot(x = df_CC['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[0],
             legend= True,
             label= f'CC Catalog',
             element='step',
             fill=False)
ax[0].set_yscale('log')
ax[0].grid(ls='--', c='gray', alpha=0.7)
ax[0].legend()
ax[0].set_ylabel('Number of earthquakes')

# Someros
filtro_datos = df_FC['Profundidad [km]'] <= profundidad_corte
df_FC_temp = df_FC[filtro_datos].copy()
sns.histplot(x = df_FC_temp['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[1],
             legend= True,
             label= f'FC Catalog',
             element='step',
             fill=False)

filtro_datos = df_BSN['Profundidad [km]'] <= profundidad_corte
df_BSN_temp = df_BSN[filtro_datos].copy()
sns.histplot(x = df_BSN_temp['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[1],
             legend= True,
             label= f'BSN Catalog',
             element='step',
             fill=False)


filtro_datos = df_CC['Profundidad [km]'] > profundidad_corte
df_CC_temp = df_CC_temp[filtro_datos].copy()
sns.histplot(x = df_CC['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[1],
             legend= True,
             label= f'CC Catalog',
             element='step',
             fill=False)
ax[1].set_yscale('log')
ax[1].grid(ls='--', c='gray', alpha=0.7)
ax[1].legend()
ax[1].set_ylabel('Number of earthquakes')

# Profundos
filtro_datos = df_FC['Profundidad [km]'] > profundidad_corte
df_FC_temp = df_FC[filtro_datos].copy()
sns.histplot(x = df_FC_temp['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[2],
             legend= True,
             label= f'FC Catalog',
             element='step',
             fill=False)

filtro_datos = df_BSN['Profundidad [km]'] > profundidad_corte
df_BSN_temp = df_BSN[filtro_datos].copy()
sns.histplot(x = df_BSN_temp['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[2],
             legend= True,
             label= f'BSN Catalog',
             element='step',
             fill=False)


filtro_datos = df_CC['Profundidad [km]'] <= profundidad_corte
df_CC_temp = df_CC_temp[filtro_datos].copy()
sns.histplot(x = df_CC['Tiempo'].dt.year,
             #log_scale = (False, True),
             bins = year_range,
             ax= ax[2],
             legend= True,
             label= f'CC Catalog',
             element='step',
             fill=False)
ax[2].set_yscale('log')
ax[2].grid(ls='--', c='gray', alpha=0.7)
ax[2].legend()
ax[2].set_ylabel('Number of earthquakes')
ax[2].set_xlabel('Time')

#ax[2].set_xlabel('')
ax[2].legend()
ax[2].grid(ls='--', c='gray', alpha=0.7)


ax[0].set_title(f'All depths')
ax[1].set_title(f'Depth $\leq{profundidad_corte}$ km')
ax[2].set_title(f'Depth $>{profundidad_corte}$ km')

ax[2].set_xticks(np.arange(start=1990, stop=2025, step=2))
ax[2].set_xlim(left=1993, right=2024)

plt.suptitle(f'Number of earthquakes per year')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = f'cantidad_eventos_anuales',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

En la gráfica de número de eventos por año, el BSN muestra menos de 100 eventos anuales para profundidades someras, lo que dificulta la estimación confiable de los parámetros de Gutenberg–Richter a escala anual. Por esta razón, los cálculos se realizarán de manera acumulativa en etapas posteriores del análisis, y no de forma anual.

# Calculos N

In [ ]:
mag_min = 0
mag_max = 10
steps_mag = 0.1

## Todo el catalogo

In [ ]:
diccionario_FC = {}
diccionario_BSN = {}
diccionario_CC = {}

In [ ]:
diccionario_FC['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_FC['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_FC['get_N']

In [ ]:
diccionario_BSN['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_BSN['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_BSN['get_N']

In [ ]:
diccionario_CC['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_CC['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_CC['get_N']

## Eventos someros

In [ ]:
diccionario_FC_somero = {}
diccionario_BSN_somero = {}
diccionario_CC_somero = {}

filtro_datos = df_FC['Profundidad [km]'] <= profundidad_corte
df_FC_temp = df_FC[filtro_datos]

filtro_datos = df_BSN['Profundidad [km]'] <= profundidad_corte
df_BSN_temp = df_BSN[filtro_datos]

filtro_datos = df_CC['Profundidad [km]'] <= profundidad_corte
df_CC_temp = df_CC[filtro_datos]

In [ ]:
diccionario_FC_somero['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_FC_temp['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_FC_somero['get_N']

In [ ]:
diccionario_BSN_somero['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_BSN_temp['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_BSN_somero['get_N']

In [ ]:
diccionario_CC_somero['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_CC_temp['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_CC_somero['get_N']

## Eventos profundos

In [ ]:
diccionario_FC_profundo = {}
diccionario_BSN_profundo = {}
diccionario_CC_profundo = {}

filtro_datos = df_FC['Profundidad [km]'] > profundidad_corte
df_FC_temp = df_FC[filtro_datos]

filtro_datos = df_BSN['Profundidad [km]'] > profundidad_corte
df_BSN_temp = df_BSN[filtro_datos]

filtro_datos = df_CC['Profundidad [km]'] > profundidad_corte
df_CC_temp = df_CC[filtro_datos]

In [ ]:
diccionario_FC_profundo['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_FC_temp['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_FC_profundo['get_N']

In [ ]:
diccionario_BSN_profundo['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_BSN_temp['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_BSN_profundo['get_N']

In [ ]:
diccionario_CC_profundo['get_N'] = sup.stats.obtener_N_LGR(magnitudes_array = df_CC_temp['Magnitud'].values,
                                                  magnitud_minima = mag_min,
                                                  magnitud_maxima = mag_max,
                                                  steps_ = steps_mag,
                                                  texto = False)

diccionario_CC_profundo['get_N']

### Grafico N

In [ ]:
fig, ax = plt.subplots(nrows=3, ncols=3, figsize=(10,10),
                       sharey='row', sharex='all')

titles_ = ['FC', 'BSN', 'CC']
# Todos
for index_, diccionario in enumerate([diccionario_FC,
                                      diccionario_BSN,
                                      diccionario_CC]):
  sup.plots.GutenbergRichter_scatter_plot(N_count = diccionario['get_N']['N'],
                                          magnitudes= diccionario['get_N']['m'],
                                          ax_h= ax[0, index_],
                                          magnitud_maxima= mag_max)
  if index_ > 0:
    ax[0, index_].set_ylabel('')
  ax[0, index_].set_xlabel('')
  ax[0, index_].set_title(f'{titles_[index_]} - All depths')

# Someros
for index_, diccionario in enumerate([diccionario_FC_somero,
                                      diccionario_BSN_somero,
                                      diccionario_CC_somero]):
  sup.plots.GutenbergRichter_scatter_plot(N_count = diccionario['get_N']['N'],
                                          magnitudes= diccionario['get_N']['m'],
                                          ax_h= ax[1, index_],
                                          magnitud_maxima= mag_max)
  if index_ > 0:
    ax[1, index_].set_ylabel('')
  ax[1, index_].set_xlabel('')
  ax[1, index_].set_title(f'{titles_[index_]} - Depth $\leq{profundidad_corte}$ km')

# Profundos
for index_, diccionario in enumerate([diccionario_FC_profundo,
                                      diccionario_BSN_profundo,
                                      diccionario_CC_profundo]):
  sup.plots.GutenbergRichter_scatter_plot(N_count = diccionario['get_N']['N'],
                                          magnitudes= diccionario['get_N']['m'],
                                          ax_h= ax[2, index_],
                                          magnitud_maxima= mag_max)
  if index_ > 0:
    ax[2, index_].set_ylabel('')
  ax[2, index_].set_xlabel('')
  ax[2, index_].set_title(f'{titles_[index_]} - Depth $>{profundidad_corte}$ km')

ax[2, 0].set_xlabel('Magnitude')
ax[2, 1].set_xlabel('Magnitude')
ax[2, 2].set_xlabel('Magnitude')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'grafico_N_m_',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

# Calculo Gutenberg Richter

Para el cálculo de los parámetros a-value y b-value se empleará la metodología Boosted 2 (donde Mc se determina mediante el criterio de máxima curvatura), dado que en las pruebas preliminares fue la que produjo resultados más cercanos a los reportados por Londoño et al. (2019). A diferencia de dicho estudio, donde se aplican filtros que eliminan una fracción importante del catálogo, en este trabajo no se han descartado eventos, salvo el criterio de disponibilidad de ML como tipo de magnitud. Esta decisión busca conservar la muestra estadística lo más completa posible en cada subcatálogo.

Además, considerando que una parte del proyecto está orientada a la prevención sísmica, se optó por incluir la mayor cantidad de eventos registrados en la zona. Aunque la calidad de algunos eventos puede estar condicionada por diversos factores (por ejemplo, la densidad y cobertura de la red de estaciones), su inclusión aporta información sobre la ocurrencia sísmica y permite evaluar el comportamiento del catálogo bajo un enfoque más conservador.

Bajo estas consideraciones, se implementará el método Boosted utilizando un esquema de remuestreo del 80% de los datos sin reemplazo.

In [ ]:
mag_min = 0
mag_max = 10
steps_mag = 0.1
porcentaje_remuestreo = 0.8
numero_remuestreos = 1000
reemplazo = False
correcion_mc = 0.0

## Todo el catalogo

In [ ]:
diccionario_FC['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_FC['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_FC['Boosted']

In [ ]:
diccionario_BSN['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_BSN['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_BSN['Boosted']

In [ ]:
diccionario_CC['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_CC['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_CC['Boosted']

## Eventos someros

In [ ]:
filtro_datos = df_FC['Profundidad [km]'] <= profundidad_corte
df_FC_temp = df_FC[filtro_datos].copy()

filtro_datos = df_BSN['Profundidad [km]'] <= profundidad_corte
df_BSN_temp = df_BSN[filtro_datos].copy()

filtro_datos = df_CC['Profundidad [km]'] <= profundidad_corte
df_CC_temp = df_CC[filtro_datos].copy()

In [ ]:
diccionario_FC_somero['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_FC_temp['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_FC_somero['Boosted']

In [ ]:
diccionario_BSN_somero['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_BSN_temp['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_BSN_somero['Boosted']

In [ ]:
diccionario_CC_somero['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_CC_temp['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_CC_somero['Boosted']

## Eventos profundos

In [ ]:
filtro_datos = df_FC['Profundidad [km]'] > profundidad_corte
df_FC_temp = df_FC[filtro_datos].copy()

filtro_datos = df_BSN['Profundidad [km]'] > profundidad_corte
df_BSN_temp = df_BSN[filtro_datos].copy()

filtro_datos = df_CC['Profundidad [km]'] > profundidad_corte
df_CC_temp = df_CC[filtro_datos].copy()

In [ ]:
diccionario_FC_profundo['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_FC_temp['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_FC_profundo['Boosted']

In [ ]:
diccionario_BSN_profundo['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_BSN_temp['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_BSN_somero['Boosted']

In [ ]:
diccionario_CC_profundo['Boosted'] = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = df_CC_temp['Magnitud'],
                                                           percentage_data = porcentaje_remuestreo, # 0.8
                                                           magnitud_minima = mag_min,
                                                           magnitud_maxima = mag_max,
                                                           steps_ = steps_mag,
                                                           n_resamples = numero_remuestreos,
                                                           replace_ = reemplazo,
                                                           correcion_m = correcion_mc)

diccionario_CC_profundo['Boosted']

## Dataframe GR

In [ ]:
estadisticos = ['mc_mean', 'mc_std',
                'b_mean', 'b_std',
                'a_mean', 'a_std']

lista_diccionarios = {'FC': diccionario_FC,
                      'BSN': diccionario_BSN,
                      'CC': diccionario_CC,
                      'FC somero': diccionario_FC_somero,
                      'BSN somero': diccionario_BSN_somero,
                      'CC somero': diccionario_CC_somero,
                      'FC profundo': diccionario_FC_profundo,
                      'BSN profundo': diccionario_BSN_profundo,
                      'CC profundo': diccionario_CC_profundo,}

diccionario_vacio = {}

for key, diccionario in lista_diccionarios.items():
  diccionario_vacio[key] = []
  for estadistico in estadisticos:
    diccionario_vacio[key].append(diccionario['Boosted'][estadistico])

df_GR_todos = pd.DataFrame(diccionario_vacio)
df_GR_todos.index = estadisticos

df_GR_todos.round(2)

In [ ]:
df_GR_todos.to_csv(path_or_buf = os.path.join(path_save_figures, 'df_GR_todos.csv'),
                   sep=';',
                   decimal=',',
                   index=True)

### Grafico GR

In [ ]:
min_max_GR = [1.0, 7.0]

fig, ax = plt.subplots(nrows=3, ncols=3, figsize=(10,10),
                       sharey='row', sharex='all')

titles_ = ['FC', 'BSN', 'CC']
# Todos
for index_, diccionario in enumerate([diccionario_FC,
                                      diccionario_BSN,
                                      diccionario_CC]):
  sup.plots.GutenbergRichter_scatter_plot(N_count = diccionario['get_N']['N'],
                                          magnitudes = diccionario['get_N']['m'],
                                          ax_h = ax[0, index_],
                                          mc = diccionario['Boosted']['mc_mean'].round(2),
                                          magnitud_maxima= mag_max)
  sup.plots.GutenbergRichter_line_plot(b_value = diccionario['Boosted']['b_mean'],
                                       a_value = diccionario['Boosted']['a_mean'],
                                       ax_h = ax[0, index_],
                                       magnitud_minima = min_max_GR[0],
                                       magnitud_maxima = min_max_GR[1])
  if index_ > 0:
    ax[0, index_].set_ylabel('')
  ax[0, index_].set_xlabel('')
  ax[0, index_].set_title(f'{titles_[index_]} - All depths')

# Someros
for index_, diccionario in enumerate([diccionario_FC_somero,
                                      diccionario_BSN_somero,
                                      diccionario_CC_somero]):
  sup.plots.GutenbergRichter_scatter_plot(N_count = diccionario['get_N']['N'],
                                          magnitudes = diccionario['get_N']['m'],
                                          ax_h = ax[1, index_],
                                          mc = diccionario['Boosted']['mc_mean'].round(2),
                                          magnitud_maxima= mag_max)
  sup.plots.GutenbergRichter_line_plot(b_value = diccionario['Boosted']['b_mean'],
                                       a_value = diccionario['Boosted']['a_mean'],
                                       ax_h = ax[1, index_],
                                       magnitud_minima = min_max_GR[0],
                                       magnitud_maxima = min_max_GR[1])
  if index_ > 0:
    ax[1, index_].set_ylabel('')
  ax[1, index_].set_xlabel('')
  ax[1, index_].set_title(f'{titles_[index_]} - Depth $\leq{profundidad_corte}$ km')

# Profundos
for index_, diccionario in enumerate([diccionario_FC_profundo,
                                      diccionario_BSN_profundo,
                                      diccionario_CC_profundo]):
  sup.plots.GutenbergRichter_scatter_plot(N_count = diccionario['get_N']['N'],
                                          magnitudes = diccionario['get_N']['m'],
                                          ax_h = ax[2, index_],
                                          mc = diccionario['Boosted']['mc_mean'].round(2),
                                          magnitud_maxima= mag_max)
  sup.plots.GutenbergRichter_line_plot(b_value = diccionario['Boosted']['b_mean'],
                                       a_value = diccionario['Boosted']['a_mean'],
                                       ax_h = ax[2, index_],
                                       magnitud_minima = min_max_GR[0],
                                       magnitud_maxima = min_max_GR[1])
  if index_ > 0:
    ax[2, index_].set_ylabel('')
  ax[2, index_].set_xlabel('')
  ax[2, index_].set_title(f'{titles_[index_]} - Depth $>{profundidad_corte}$ km')

ax[2, 0].set_xlabel('Magnitude')
ax[2, 1].set_xlabel('Magnitude')
ax[2, 2].set_xlabel('Magnitude')

plt.tight_layout()

sup.plots.save_plot(path_save_figuras = path_save_figures,
                    file_name = 'grafico_GR_boosted_',
                    formato = 'png',
                    dpi_ = 300)

plt.show()

# Calculo GR acumulado

Se procederá a calcular los parámetros a-value, b-value y Mc de forma acumulativa. Para ello, el análisis se iniciará en el período mínimo 1993–1994 y, posteriormente, se irá incorporando información de manera progresiva, añadiendo un año a la vez, hasta completar el intervalo 1993–2024.

In [ ]:
print('Año mínimo: ', df_FC['Tiempo'].dt.year.min())
print('Año máximo: ', df_FC['Tiempo'].dt.year.max())

In [ ]:
year_range = np.arange(df_FC['Tiempo'].dt.year.min(),
                       df_FC['Tiempo'].dt.year.max()+1)
year_range

## Todo el catalogo

In [ ]:
diccionario_Acumulado_todos = {}

estadisticos = ['mc_mean', 'mc_std',
                'b_mean', 'b_std',
                'a_mean', 'a_std']

catalogos_ = {'FC': df_FC,
              'BSN': df_BSN,
              'CC': df_CC}


for key, catalogo in catalogos_.items():
  print('\n')
  lista_years = []
  lista_estadisticos_label = []
  lista_estadisticos = []
  print(key)
  catalogo_temp = catalogo.copy()
  catalogo_temp.reset_index(inplace=True, drop=True)
  for i in range(1, len(year_range)):
    years_ = [year_range[0], year_range[i]]
    print(years_)
    filtro_datos = (catalogo_temp['Tiempo'].dt.year >= years_[0]) & (catalogo_temp['Tiempo'].dt.year <= years_[1])
    #filtro_datos = catalogo_temp['Tiempo'].dt.year.isin(year_range[0:i+1])
    print(year_range[0], year_range[i])
    magnitudes_tempo = catalogo_temp['Magnitud'][filtro_datos]
    print('Suma filtro', filtro_datos.sum())
    print('Shape magnitudes', magnitudes_tempo.shape)
    diccionario_temp = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = magnitudes_tempo,
                                                            percentage_data = porcentaje_remuestreo, # 0.8
                                                            magnitud_minima = mag_min,
                                                            magnitud_maxima = mag_max,
                                                            steps_ = steps_mag,
                                                            n_resamples = numero_remuestreos,
                                                            replace_ = reemplazo,
                                                            correcion_m = correcion_mc)

    for estadistico in estadisticos:
      lista_years.append(f'{year_range[0]}-{year_range[i]}')
      lista_estadisticos_label.append(estadistico)
      lista_estadisticos.append(diccionario_temp[estadistico])


  diccionario_Acumulado_todos[key] = {'Años': lista_years,
                                      'Estadistico': lista_estadisticos_label,
                                      'Valor': lista_estadisticos}

### Dataframe

In [ ]:
diccionario_Acumulado_todos.keys()

In [ ]:
diccionario_Acumulado_todos['FC'].keys()

In [ ]:
pd.DataFrame(diccionario_Acumulado_todos['FC']).head()

In [ ]:
diccionario_df_acumulados = {}

for key, diccionario in diccionario_Acumulado_todos.items():
  df_temp = pd.DataFrame(diccionario)
  estadisticos = df_temp['Estadistico'].unique()

  df_vacio = pd.DataFrame(columns = ['Años'])
  df_vacio['Años'] = df_temp['Años'].unique()
  df_vacio['Catalogo'] = key
  #df_vacio.set_index('Años', inplace=True)

  for estadistico in estadisticos:
    df_vacio[estadistico] = df_temp[df_temp['Estadistico'] == estadistico]['Valor'].values

  diccionario_df_acumulados[key] = df_vacio


In [ ]:
diccionario_df_acumulados.keys()

In [ ]:
diccionario_df_acumulados['FC']

In [ ]:
for key, df in diccionario_df_acumulados.items():
  df.to_csv(path_or_buf = os.path.join(path_save_figures, f'df_GR_acumulado_{key}_todos.csv'),
            sep=';',
            decimal=',',
            index=True)

## Eventos someros

In [ ]:
diccionario_Acumulado_someros = {}

estadisticos = ['mc_mean', 'mc_std',
                'b_mean', 'b_std',
                'a_mean', 'a_std']

catalogos_ = {'FC somero': df_FC,
              'BSN somero': df_BSN,
              'CC somero': df_CC}


for key, catalogo in catalogos_.items():
  print('\n')
  lista_years = []
  lista_estadisticos_label = []
  lista_estadisticos = []
  filtro_datos = catalogo['Profundidad [km]'] <= profundidad_corte
  catalogo_temp = catalogo[filtro_datos].copy()
  catalogo_temp.reset_index(inplace=True, drop=True)
  print(key)
  for i in range(1, len(year_range)):
    years_ = [year_range[0], year_range[i]]
    print(years_)
    filtro_datos = (catalogo_temp['Tiempo'].dt.year >= years_[0]) & (catalogo_temp['Tiempo'].dt.year <= years_[1])
    #filtro_datos = catalogo_temp['Tiempo'].dt.year.isin([year_range[0], year_range[i]])
    print(year_range[0], year_range[i])
    magnitudes_tempo = catalogo_temp['Magnitud'][filtro_datos]
    print('Suma filtro', filtro_datos.sum())
    print('Shape magnitudes', magnitudes_tempo.shape)
    diccionario_temp = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = magnitudes_tempo,
                                                            percentage_data = porcentaje_remuestreo, # 0.8
                                                            magnitud_minima = mag_min,
                                                            magnitud_maxima = mag_max,
                                                            steps_ = steps_mag,
                                                            n_resamples = numero_remuestreos,
                                                            replace_ = reemplazo,
                                                            correcion_m = correcion_mc)

    for estadistico in estadisticos:
      lista_years.append(f'{year_range[0]}-{year_range[i]}')
      lista_estadisticos_label.append(estadistico)
      lista_estadisticos.append(diccionario_temp[estadistico])

  diccionario_Acumulado_someros[key] = {'Años': lista_years,
                                      'Estadistico': lista_estadisticos_label,
                                      'Valor': lista_estadisticos}

### Dataframe

In [ ]:
diccionario_Acumulado_someros.keys()

In [ ]:
diccionario_Acumulado_someros['FC somero'].keys()

In [ ]:
pd.DataFrame(diccionario_Acumulado_someros['FC somero']).head()

In [ ]:
diccionario_df_acumulados_somero = {}

for key, diccionario in diccionario_Acumulado_someros.items():
  df_temp = pd.DataFrame(diccionario)
  estadisticos = df_temp['Estadistico'].unique()

  df_vacio = pd.DataFrame(columns = ['Años'])
  df_vacio['Años'] = df_temp['Años'].unique()
  df_vacio['Catalogo'] = key
  #df_vacio.set_index('Años', inplace=True)

  for estadistico in estadisticos:
    df_vacio[estadistico] = df_temp[df_temp['Estadistico'] == estadistico]['Valor'].values

  diccionario_df_acumulados_somero[key] = df_vacio

In [ ]:
diccionario_df_acumulados_somero.keys()

In [ ]:
diccionario_df_acumulados_somero['FC somero']

In [ ]:
for key, df in diccionario_df_acumulados_somero.items():
  df.to_csv(path_or_buf = os.path.join(path_save_figures, f'df_GR_acumulado_{key}_.csv'),
            sep=';',
            decimal=',',
            index=True)

## Eventos profundos

In [ ]:
diccionario_Acumulado_profundo = {}

estadisticos = ['mc_mean', 'mc_std',
                'b_mean', 'b_std',
                'a_mean', 'a_std']

catalogos_ = {'FC profundo': df_FC,
              'BSN profundo': df_BSN,
              'CC profundo': df_CC}


for key, catalogo in catalogos_.items():
  print('\n')
  lista_years = []
  lista_estadisticos_label = []
  lista_estadisticos = []
  filtro_datos = catalogo['Profundidad [km]'] > profundidad_corte
  catalogo_temp = catalogo[filtro_datos].copy()
  catalogo_temp.reset_index(inplace=True, drop=True)
  print(key)
  for i in range(1, len(year_range)):
    years_ = [year_range[0], year_range[i]]
    print(years_)
    filtro_datos = (catalogo_temp['Tiempo'].dt.year >= years_[0]) & (catalogo_temp['Tiempo'].dt.year <= years_[1])
    #filtro_datos = catalogo_temp['Tiempo'].dt.year.isin([year_range[0], year_range[i]])
    print(year_range[0], year_range[i])
    magnitudes_tempo = catalogo_temp['Magnitud'][filtro_datos]
    print('Suma filtro', filtro_datos.sum())
    print('Shape magnitudes', magnitudes_tempo.shape)
    diccionario_temp = sup.stats.get_GutenbergRichter_values_boosted_2(magnitudes = magnitudes_tempo,
                                                            percentage_data = porcentaje_remuestreo, # 0.8
                                                            magnitud_minima = mag_min,
                                                            magnitud_maxima = mag_max,
                                                            steps_ = steps_mag,
                                                            n_resamples = numero_remuestreos,
                                                            replace_ = reemplazo,
                                                            correcion_m = correcion_mc)

    for estadistico in estadisticos:
      lista_years.append(f'{year_range[0]}-{year_range[i]}')
      lista_estadisticos_label.append(estadistico)
      lista_estadisticos.append(diccionario_temp[estadistico])

  diccionario_Acumulado_profundo[key] = {'Años': lista_years,
                                      'Estadistico': lista_estadisticos_label,
                                      'Valor': lista_estadisticos}

### Dataframe

In [ ]:
diccionario_Acumulado_profundo.keys()

In [ ]:
diccionario_Acumulado_profundo['FC profundo'].keys()

In [ ]:
pd.DataFrame(diccionario_Acumulado_profundo['FC profundo']).head()

In [ ]:
diccionario_df_acumulados_profundo = {}

for key, diccionario in diccionario_Acumulado_profundo.items():
  df_temp = pd.DataFrame(diccionario)
  estadisticos = df_temp['Estadistico'].unique()

  df_vacio = pd.DataFrame(columns = ['Años'])
  df_vacio['Años'] = df_temp['Años'].unique()
  df_vacio['Catalogo'] = key
  #df_vacio.set_index('Años', inplace=True)

  for estadistico in estadisticos:
    df_vacio[estadistico] = df_temp[df_temp['Estadistico'] == estadistico]['Valor'].values

  diccionario_df_acumulados_profundo[key] = df_vacio

In [ ]:
diccionario_df_acumulados_profundo.keys()

In [ ]:
diccionario_df_acumulados_profundo['FC profundo']

In [ ]:
for key, df in diccionario_df_acumulados_profundo.items():
  df.to_csv(path_or_buf = os.path.join(path_save_figures, f'df_GR_acumulado_{key}_.csv'),
            sep=';',
            decimal=',',
            index=True)

## Grafico GR acumulado

In [ ]:
diccionario_df_acumulados.keys()

In [ ]:
diccionario_df_acumulados_somero.keys()

In [ ]:
diccionario_df_acumulados_profundo.keys()

In [ ]:
variables = {'b-value': ['b_mean', 'b_std'],
             'a-value': ['a_mean', 'a_std'],
             'Mc': ['mc_mean', 'mc_std']}

for var_, estadistico_ in variables.items():
  fig, ax = plt.subplots(ncols=1, nrows=3, figsize=(7,9), sharex=True)

  # Todos
  for catalogo_, dataFrame_ in diccionario_df_acumulados.items():
    ax[0].errorbar(x = year_range[1:],
                   y = dataFrame_[estadistico_[0]],
                   yerr = dataFrame_[estadistico_[1]],
                   fmt = '-o',
                   label = catalogo_[0:3])
  ax[0].set_xlabel('')
  ax[0].legend()
  ax[0].set_ylabel(f'{var_}')
  ax[0].grid(ls='--', c='gray', alpha=0.7)

  # Someros
  for catalogo_, dataFrame_ in diccionario_df_acumulados_somero.items():
    ax[1].errorbar(x = year_range[1:],
                   y = dataFrame_[estadistico_[0]],
                   yerr = dataFrame_[estadistico_[1]],
                   fmt = '-o',
                   label = catalogo_[0:3])
  ax[1].set_xlabel('')
  ax[1].legend()
  ax[1].set_ylabel(f'{var_}')
  ax[1].grid(ls='--', c='gray', alpha=0.7)

  # Profundos
  for catalogo_, dataFrame_ in diccionario_df_acumulados_profundo.items():
    ax[2].errorbar(x = year_range[1:],
                   y = dataFrame_[estadistico_[0]],
                   yerr = dataFrame_[estadistico_[1]],
                   fmt = '-o',
                   label = catalogo_[0:3])
  #ax[2].set_xlabel('')
  ax[2].legend()
  ax[2].set_ylabel(f'{var_}')
  ax[2].grid(ls='--', c='gray', alpha=0.7)


  ax[0].set_title(f'All depths')
  ax[1].set_title(f'Depth $\leq{profundidad_corte}$ km')
  ax[2].set_title(f'Depth $>{profundidad_corte}$ km')

  plt.tight_layout()

  sup.plots.save_plot(path_save_figuras = path_save_figures,
                      file_name = f'GR_{var_}_acumulado',
                      formato = 'png',
                      dpi_ = 300)

  plt.show()
  print('\n')

# Fin